inception-external-search-elastic/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/elastic/ElasticSearchProvider.java

```diff
@@ -126,9 +137,49 @@ public boolean isConnected()
                 }
             }
             if (hit.getHighlight() != null) {
+
+                // Highlights from elastic search are small sections of the document text
+                // with the keywords surrounded by the <em> tags. There are no offset information
+                // for the highlights or the keywords in the document text. There is a feature
+                // request for it (https://github.com/elastic/elasticsearch/issues/5736).
+                // Until this feature is implemented, we currently try to find the keywords offsets
+                // by finding the matching highlight in the document text, then the keywords offset
+                // within highlight using <em> tags.
+                String originalText = hit.get_source().getDoc().getText();
+
                 // There are highlights, set them in the result
-                List<String> highlights = new ArrayList<>();
-                highlights.add(hit.getHighlight().getDoctext().get(0));
+                List<ExternalSearchHighlight> highlights = new ArrayList<>();
+                for (String highlight : hit.getHighlight().getDoctext()) {
+
+                    // remove markers from the highlight
+                    String highlight_clean = highlight.replace(highlight_start_tag, "")
+                        .replace(highlight_end_tag, "");
+
+                    // find the matching highlight offset in the original text
+                    int highlight_start_index = originalText.indexOf(highlight_clean);
+
+                    // find offset to all keywords in the highlight
+                    // they are enclosed in <em> </em> tags in the highlight
+                    String highlightTemp = highlight;
+                    ArrayList<OffsetSpan> offsets = new ArrayList<>();
```

Comment:
Variable type should be `List<OffsetSpan>`.

Reply:
Changed to `List`